doc/tag_filter.py

```diff
@@ -4,26 +4,36 @@
 import os
 
 file_name = sys.argv[1]
-file_obj = open(file_name, 'r')
+tag_comment = "@content_tag{"
+match_found = False
+prev_line = list()
+buff = list()
 
-tag_comment = "content_tag{"
-tag_match = 0
+# dedent doxygen tag documentation
+with open(file_name, 'r') as f:
+    # sys.stderr.write("checking file: " + file_name + '\n')
+    for line in f:
+        # doxygen wont document this alias if it is indented
+        if tag_comment in line:
+            ln_void, ln_tag, ln_content = line.partition(tag_comment)
+            prev_line.append('## ' + tag_comment + ln_content.rstrip('\n'))
+            match_found = True
+        elif prev_line:
+            # catch subsequent comment lines to allow for multi-line in python
+            if '#' in line:
+                ln_void, ln_tag, ln_content = line.partition('#')
+                prev_line.append(ln_content.rstrip('\n'))
+                buff.append('# \n')  # must replace with same number of lines
+            else:
+                # finally finish this line
+                buff.append(' '.join(prev_line) + '\n')
+                prev_line = []
+        else:
+            buff.append(line + '\n')
 
-# check for match
-line = file_obj.readline()
-while line and not tag_match:
-    if tag_comment in line:
-        tag_match = 1
-    line = file_obj.readline()
-file_obj.close()
-
-# output file
-file_obj = open(file_name, 'r')
-line = file_obj.readline()
-while line and tag_match:
-    # doxygen wont document this alias if it is indented
-    if tag_comment in line:
-        line = line.strip() + '\n'
-    sys.stdout.write(line)
-    line = file_obj.readline()
-file_obj.close()
+# Output the entire file if it contains a match
+# entire file is needed for context
+if match_found:
```

Comment:
The purpose of `match_found` is to not output the buffer when there is no match.
Unless we want to document all python code within the cpp api?